src/tests/ftest/container/snapshot_functionality_test.py

```diff
@@ -0,0 +1,539 @@
+                   akey. Compare it to the originally written data.
+                (4)List the snapshot and make sure it reflects the original
+                   epoch.
+                   ==>Repeat step(1) to step(4) for multiple snapshot tests.
+                (5)Verify the snapshots data.
+                (6)Destroy the snapshot.
+                (7)Check if still able to Open the destroyed snapshot and
+                   Verify the snapshot removed from the snapshot list.
+        Use Cases: Combinations with minimun 1 client and 1 server.
+        :avocado: tags=snap,snapshotinfo
+        """
+
+        coh_list = []
+        container_epoch_list = []
+        snapshot_list = []
+        test_data = []
+        snapshot_index = 1
+        snapshot_loop = 10
+        obj_cls = self.params.get("obj_class", '/run/object_class/*')
+        akey = self.params.get("akey", '/run/snapshot/*', default="akey")
+        dkey = self.params.get("dkey", '/run/snapshot/*', default="dkey")
+        rand_str = lambda n: ''.join([random.choice(string.lowercase)
+                                      for i in xrange(n)])
+        #
+        #Test loop for creat, modify and snapshot object in the DAOS container.
+        #
+        while snapshot_index <= snapshot_loop:
+            #(1)Create an object, write some data into it, and take a snapshot
+            #size = random.randint(1, 100) + 1
+            thedata = "--->>>Happy Daos Snapshot Testing " + \
+                str(snapshot_index) + \
+                "<<<---" + rand_str(random.randint(1, 100)+1)
+            datasize = len(thedata) + 1
+            try:
+                obj, epoch = self.container.write_an_obj(
+                    thedata, datasize, dkey, akey, obj_cls=obj_cls)
+                obj.close()
+            except DaosApiError as error:
+                self.fail(
+                    "##Test failed during the initial object write: %s"
+                    , str(error))
+            #Take a snapshot of the container
+            snapshot = DaosSnapshot(self.context)
+            snapshot.create(self.container.coh, epoch)
+            self.log.info("==Wrote an object and created a snapshot")
+
+            #Display snapshot
+            substep = "1." + str(snapshot_index)
+            self.log.info("==(1)Test step %s", substep)
+            self.log.info("==self.container epoch=     %s", epoch)
+            self.log.info("==snapshot.epoch= %s"
+                          , snapshot.epoch)
+            self.display_snapshot(snapshot)
+
+            #Save snapshot test data
+            coh_list.append(self.container.coh)
+            container_epoch_list.append(epoch)
+            snapshot_list.append(snapshot)
+            test_data.append(thedata)
+            snapshot_index += 1
+
+            #(2)Make changes to the data object. The write_an_obj function does
+            #   a commit when the update is complete
+            more_transactions = 100
```

Comment:
sorry, don't get what those transactions are about.

Reply:
As discussed in the review meeting.